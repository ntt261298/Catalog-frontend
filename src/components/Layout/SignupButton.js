import React from 'react';
import { connect } from 'react-redux';
import { PropTypes } from 'prop-types';
import { showSignup } from '../../actions/app';

export const SignupButton = props => (
  <button type="button" onClick={props.showSignup} className="btn-signup">Signup</button>
);

SignupButton.propTypes = {
  showSignup: PropTypes.func.isRequired,
};

export default connect(null, { showSignup })(SignupButton);
