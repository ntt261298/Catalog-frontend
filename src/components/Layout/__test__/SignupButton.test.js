import React from 'react';
import { shallow } from '../../../enzyme';
import { SignupButton } from '../SignupButton';

describe('src/Layout/SignupButton.js', () => {
  let wrapper;
  let props;
  let signupButton;
  const update = () => {
    signupButton = wrapper.find('button');
  };

  const setup = () => {
    wrapper = shallow(
      <SignupButton {...props} />,
    );
    update();
  };

  beforeEach(() => {
    props = {
      showSignup: jest.fn(),
    };
  });

  it('should render signup button', () => {
    setup();
    // Expect the wrapper object to be defined
    expect(wrapper).toMatchSnapshot();
  });

  it('should call showSignup when signup button is clicked', () => {
    setup();
    signupButton.simulate('click');
    expect(props.showSignup).toHaveBeenCalled();
  });
});
