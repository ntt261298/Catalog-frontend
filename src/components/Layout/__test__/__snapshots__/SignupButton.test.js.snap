// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Layout/SignupButton should render signup button 1`] = `
<button
  className="btn-signup"
  onClick={[Function]}
  type="button"
>
  Signup
</button>
`;
